src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -4363,11 +4385,15 @@ void MacroAssembler::_verify_oop(Register reg, const char* s, const char* file,
 }
 
 void MacroAssembler::vallones(XMMRegister dst, int vector_len) {
-  if (UseAVX > 2 && (vector_len == Assembler::AVX_512bit || VM_Version::supports_avx512vl())) {
+  // vpcmpeqd has special dependency treatment so it should be preferred to vpternlogd
```

Comment:
Comment is not clear.

Reply:
I have remeasured the statement, it seems that only the non-vex encoding version receives the special dependency treatment, so I reverted this change and add a comment for clarification.

The optimisation can be found noticed in [The microarchitecture of Intel, AMD and VIA CPUs: An optimization guide for assembly programmers and compiler makers](https://www.agner.org/optimize/) on several architectures such as in section 9.8 (Register allocation and renaming in Sandy Bridge and Ivy Bridge pipeline).

I have performed measurements on uica.uops.info . While this sequence gives 1.37 cycles/iteration on Skylake and Icelake

    pcmpeqd xmm0, xmm0
    paddd xmm0, xmm1
    paddd xmm0, xmm1
    paddd xmm0, xmm1

This version has the throughput of 4 cycles/iteration

    vpcmpeqd xmm0, xmm0, xmm0
    vpaddd xmm0, xmm1, xmm0
    vpaddd xmm0, xmm1, xmm0
    vpaddd xmm0, xmm1, xmm0

Which indicates the `vpcmpeqd` failing to break dependencies on `xmm0` as opposed to the `pcmpeqd` instruction.

Thanks.

Reply:
Both the above JIT sequences have true dependency chain,  there is no scope of any additional architecture imposed false dependency doing any further perf degradation for which we use dep-breaking idioms.

Reply:
I'm sorry I don't quite understand what do you mean here, what I meant is that while `pcmpeqd xmmk, xmmk` is a dep-breaking idiom, `vpcmpeqd xmmk, xmmk, xmmk` seems to not be. As a result, I reverted that change and in this context, the only change is I added a branch for non-AVX machines. Please have a review for this patch. Thank you very much.

Reply:
Yes, its a valid one-idiom and as per section E.1.2 of [X86 Optimization manual](https://cdrdv2.intel.com/v1/dl/getContent/671488)  such idioms are resolved by renamer and does not reach execution ports. 

I faintly remember that there was a subtle difference b/w handling of zeroing/one idioms on certain targets where in some cases one-idioms still go beyond renamer. But, we can keep this change of your since even if all-one idiom (vpcmpeqd) reach execution port, latency wise it same a vpternlog over 256 bit vector.